documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,79 @@
+Strimzi supports the SASL_OAUTHBEARER and SASL_PLAIN mechanisms for OAuth 2.0 authentication. 
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+The authentication flow between clients, the authorization server, and Kafka brokers is different for each mechanism.
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+Configure clients to use SASL_OAUTHBEARER for authentication with Kafka brokers whenever possible. 
+This mechanism provides a higher level of security because client credentials are _never_ shared with Kafka brokers. 
+Consider using SASL_PLAIN only with Kafka clients that do not support SASL_OAUTHBEARER. 
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+.SASL_OAUTHBEARER overview
+
+Support for SASL_OAUTHBEARER is built in to many Kafka clients. 
+However, the developer must often write custom callbacks that provide access tokens for authentication, as needed.
+Strimzi provides these callbacks for standard Java Kafka clients (clients that use the Kafka producer and consumer APIs). 
+Therefore, developers do not need to provide custom callbacks in their application code. 
+
+With SASL_OAUTHBEARER, the client initiates a session with the Kafka broker for credentials exchange, where credentials take the form of a bearer token provided by the Strimzi OAuth callback code. 
+Using the callbacks, you can configure token provision in one of three ways:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+* A long-lived refresh token, obtained manually at configuration time
+
+SASL_OAUTHBEARER is automatically enabled in the `oauth` listener configuration for the Kafka broker. 
+You can set the `enableOauthBearer` property to `true`, though this is not required.
+
+[source,yaml,subs="attributes+"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    enableOauthBearer: true
+----
+
+.SASL_PLAIN overview
+
+SASL_PLAIN, also referred to as _OAuth 2.0 over PLAIN_, is a simple authentication mechanism supported by all Kafka client tools (including developer tools such as kafkacat). 
+Client credentials are handled centrally, behind an authorization server that complies with OAuth 2.0 and OpenID Connect (OIDC).
+
+When used with the provided callbacks, SASL_PLAIN allows Kafka clients to authenticate with Kafka brokers using either of the following methods:
+
+* OAuth access token
+
+* OAuth `clientId` and `secret` (the client credentials)
+
+The client must enable SASL_PLAIN and set the `username` and `password`. 
+If the client prefixes the `password` with `$accessToken:` followed by the value of the access token (access token ???), the authorization server will interpret the password as the access token. 
```

Comment:
I guess the `(access token ???)`  should not be here?

Reply:
Sorry, that was my mistake.

Reply:
No worries ... that is what reviews are for ;-)